### PR TITLE
Adding `p` as timezone offset, e.g. `+02:00`

### PR DESCRIPTION
**Pull request by fboes**
Because the feed validator for ATOM complains about timezone offset being given as `+0200` and not as `+02:00`, I added a slight modification of the `o` and called it `p` (like in PHP's date function).

I ended up killing my original pull request at https://github.com/felixge/node-dateformat/pull/56 (please close that) and building a completely new fix.

Cheers & thanks for this great lib,
Frank

closes: #148
closes: #81

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -126,7 +127,7 @@ dateFormat(now, "N");
 | `longTime`        | `h:MM:ss TT Z`                 | 5:46:21 PM EST           |
 | `isoDate`         | `yyyy-mm-dd`                   | 2007-06-09               |
 | `isoTime`         | `HH:MM:ss`                     | 17:46:21                 |
-| `isoDateTime`     | `yyyy-mm-dd'T'HH:MM:ss`        | 2007-06-09T17:46:21      |
+| `isoDateTime`     | `yyyy-mm-dd'T'HH:MM:sso`       | 2007-06-09T17:46:21+0700 |
```
Comment (chase-manning) [inline Readme.md:130]: @fboes The isoDateTime mask is not changing as part of this, so the documentation for this should stay the same. Please revert this change :smile:

File: test/test_mask-o.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+var assert = require("assert");
+var dateFormat = require("../lib/dateformat");
+
+describe("Mask: 'o'", function () {
+  it("should get timezone for any date as smoething like [+-]XXXX", function (done) {
```
Comment (chase-manning) [inline test/test_mask-o.js:5]: @fboes Spelling error `smoething` should be `something`

File: test/test_mask-p.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+var assert = require("assert");
+var dateFormat = require("../lib/dateformat");
+
+describe("Mask: 'p'", function () {
+  it("should get timezone for any date as smoething like [+-]XX:XX", function (done) {
```
Comment (chase-manning) [inline test/test_mask-p.js:5]: @fboes Spelling error `smoething` should be `something`

**Comment by chase-manning** [comment]
@fboes Yay! Thanks dude, awesome PR, great new feature! Have added some comments, once those resolved and the CI build passes, will merge and release a new version to npm :smile:

File: Readme.md (hunk 1 above)
Comment (chase-manning) [reply Readme.md:130]: Oh nevermind, the mask actually includes the `o` and you corrected an error in the documentation 😲  Nice job! :smile:

Comment (fboes) [reply Readme.md:130]: @chase-manning You are right, my fix does not introduce this behaviour. Instead I believe that it is already there, but not documented,

Using `isoDateTime` I already get a result like `2020-12-05T10:13:29+0100`, and the code of the lib also states it to be `yyyy-mm-dd'T'HH:MM:sso` - so I thought to fix the documentation as well. ;)

```javascript
  dateFormat.masks = {
    default: "ddd mmm dd yyyy HH:MM:ss",
    shortDate: "m/d/yy",
    paddedShortDate: "mm/dd/yyyy",
    mediumDate: "mmm d, yyyy",
    longDate: "mmmm d, yyyy",
    fullDate: "dddd, mmmm d, yyyy",
    shortTime: "h:MM TT",
    mediumTime: "h:MM:ss TT",
    longTime: "h:MM:ss TT Z",
    isoDate: "yyyy-mm-dd",
    isoTime: "HH:MM:ss",
    isoDateTime: "yyyy-mm-dd'T'HH:MM:sso",
    isoUtcDateTime: "UTC:yyyy-mm-dd'T'HH:MM:ss'Z'",
    expiresHeaderFormat: "ddd, dd mmm yyyy HH:MM:ss Z",
  };
```

But if you want me to I remove that from the pull request, as it is not linked with the rest of the pull request.

File: test/test_mask-p.js (hunk 3 above)
Comment (fboes) [reply test/test_mask-p.js:5]: I fixed that spelling mistake. It's one of my favourite typos early in the morning without a coffee. :)

File: Readme.md (hunk 1 above)
Comment (chase-manning) [reply Readme.md:130]: Yeah you were right! The docs were incorrect and you fixed them :smile:

**Comment by fboes** [comment]
@chase-manning Do I have to open a new pull request to fix my typos?

**Comment by chase-manning** [comment]
@fboes I merged anyway as not a big deal :smile: If you want to create another PR for the typos then got for it, but all good if not.
